File: Julia/rPowerLaw/rPowerLaw.jl
using Random, Distributions, Plots

function GetForOneSample(xmin,alpha)
    u = rand(Uniform(0,1))
    return xmin*((1-u)^(1/(1-alpha)))
end

function generate(n, xmin, alpha)
    return [GetForOneSample(xmin, alpha) for i in 1:n]
end

# Testing
histogram(generate(n = 1000, xmin = 2, alpha = 2.5))

File: Python/rpowerlaw/rpowerlaw.py
import numpy as np

def generate(n, xmin, alpha):
    """
    Generate/Draw powerlaw samples for Monte-Carlo process, powerlaw analysis, etc.
    :param n: number of samples to be generated
    :param xmin: xmin of the population
    :param alpha: alpha parameter of the population
    """
    s = []
    for _ in range(n):
        s.append(xmin*((1-np.random.uniform(0,1))**(1/(1-alpha))))
    return s

def getCumulative(n, x, xmin, alpha):
    """
    Get CDF of powerlaw at x
    :param n: number of samples to be generated
    :param x: x at which CDF needs to be calculated
    :param xmin: xmin of the population
    :param alpha: alpha parameter of the population
    """
    return (1-((x/xmin)**(1-alpha)))